#!/usr/bin/env python

PACKAGE = "dji_m100_trajectory"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       	Name		Type      	Level	Description			Default 	Min	Max
# Noise levels
gen.add(	"start_rec",	bool_t,		0,	"Start recording flag", 	False)
gen.add(	"stop_rec",	bool_t,		0,	"Stop recording flag", 	 	False)
gen.add( 	"pred_horizon",	double_t, 	0, 	"NMPC's prediction horizon size",30, 		1, 	50)
rec_config_enum = gen.enum([ 	gen.const("gp", int_t, 0, "Record configuration for GP)"),
				gen.const("nmhe",int_t, 1, "Record configuration for NMHE"),],
                     		"An enum to set record configuration")
gen.add(	"rec_config", 		int_t, 		0, 	"Record configuration", 			0, 	0, 	1, edit_method=rec_config_enum)
gen.add(	"rec_Fx_dist",	bool_t,		0,	"Record Fx disturbance data", 	True)
gen.add(	"rec_Fy_dist",	bool_t,		0,	"Record Fy disturbance data", 	True)
gen.add(	"rec_Fz_dist",	bool_t,		0,	"Record Fz disturbance data", 	True)
gen.add( 	"max_rec_time",	double_t, 	0, 	"Max recording time [sec]", 	1000, 	0, 	6000)

exit(gen.generate(PACKAGE, "recdata", "set_recdata"))
